src/adagrams.js

```diff
@@ -1,8 +1,150 @@
+
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const letters = {
+      "A": 9,
+      "N": 6,
+      "B": 2,
+      "O": 8,
+      "C": 2,
+      "P": 2,
+      "D": 4,
+      "Q": 1, 
+      "E": 12,
+      "R": 6,
+      "F": 2,
+      "S": 4,
+      "G": 3,
+      "T": 6,
+      "H": 2,
+      "U": 4,
+      "I": 9,
+      "V": 2,
+      "J": 1,
+      "W": 2,
+      "K": 1,
+      "X": 1,
+      "L": 4,
+      "Y": 2,
+      "M": 2,	
+      "Z": 1,
+    };
+
+    const lettersArray = Object.keys(letters);
+    const newArray = [];
+
+    lettersArray.forEach(function(char) {
+      let i = 0;
+      while (i < letters[char]) {
+        newArray.push(char);
+        i++;
+      }
+    });
+  
+    let i = 0;
+    const hand = [];
+
+    while (i < 10) {
+      let randomLocation = Math.floor(newArray.length * Math.random());
+      let currentLetter = newArray[randomLocation];
+      hand.push(currentLetter);
+      newArray.splice(randomLocation, 1);
+      i++;
+    };
+      return hand;
+    }, 
+
+  usesAvailableLetters(word, drawn) {
+
+    const drawnObject = {};
+    
+    drawn.forEach((letter) => {
+      if (drawnObject[letter]) {
+        drawnObject[letter] += 1;
+      }
+      else {
+        drawnObject[letter] = 1;
+      }
+    });
+
+    const wordArray = word.split('');
+    let i = 0
+    while (i < wordArray.length) {
+      let theLetter = wordArray[i];
+    
+      if (drawnObject[theLetter] > 0) {
+      drawnObject[theLetter] -= 1;
+      }
+      else {
+        return false;
+      }
+      i++;
+    }
+
+    return true;
   },
+  scoreWord(word) {
+    // build the scorechart
+    const ones = ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'];
```

Comment:
Consider using an object to DRY up your code so that you can use the values to update your score. This would remove all the work you're doing with the `.forEach` loops below.

`
{
A: 1,
E: 1,
I: 1, 
...
}
`